design/resource-management.md

```diff
@@ -0,0 +1,329 @@
+      - [In Kubernetes  + CRI-O (v1 shim)](#in-kubernetes----cri-o--v1-shim-)
+    + [Issues with current implementation](#issues-with-current-implementation)
+      - [Accurate usage accounting](#accurate-usage-accounting)
+      - [Node stability](#node-stability)
+      - [Consistent guaranteed pod behavior](#consistent-guaranteed-pod-behavior)
+      - [OOM, unbound CPU utilization](#oom--unbound-cpu-utilization)
+  * [Proposed Changes](#proposed-changes)
+    + [Summary](#summary)
+    + [Details](#details)
+      - [Pod Sandbox Cgroup](#pod-sandbox-cgroup)
+  * [Alternatives Considered](#alternatives-considered)
+    + [Only constrain vCPUs, leaving remaining threads for system reserved](#only-constrain-vcpus--leaving-remaining-threads-for-system-reserved)
+  * [Opens](#opens)
+    + [static CPU configurations](#static-cpu-configurations)
+
+## Background
+
+With 1.6 release of Kata Containers there are some issues with resource management resulting
+in inconsistent behavior. This document descibes the state of 1.6, and a suggested implementation
+for 1.7 version of Kata.
+
+Before diving into the gaps and behavior exhibited in Kata Containers, it is important to have
+a thorough understanding of how cgroups are leveraged by Kubernetes.  It is both straight forward
+and confusing.  An in-depth guide is available for background in [mcastelino's gist](https://gist.github.com/mcastelino/b8ce9a70b00ee56036dadd70ded53e9f).
+This may be good content to include in our repository eventually, or as part of the Kata blog series,
+but let's leave it out of the scope of this initial documentation.
+
+This document is part of the `cgroup-sprint` GitHub milestone, and can be observed in the milestone's
+[GitHub project](https://github.com/orgs/kata-containers/projects/17).
+
+
+## Existing Behavior (Kata 1.6):
+
+### Behavior Observed using various upper layer tools
+
+To exhibit current behavior, we utilize a simple guaranteed pod description:
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: guar-runc
+spec:
+  containers:
+  - name: cont-2cpu-400m
+    image: busybox
+    resources:
+      limits:sadf
+        cpu: 2
+        memory: "400Mi"
+    command: ["md5sum"]
+    args: ["/dev/urandom"]
+  - name: cont-3cpu-200m
+    image: busybox
+    resources:
+      limits:
+        cpu: 3
+        memory: "200Mi"
+    command: ["md5sum"]
+    args: ["/dev/urandom"]
+```
+We'll show the behavior starting with the simplest scenario, Docker, followed by containerd
+and CRI-O.
+
+#### In Docker
+
+#### In Kubernetes + Containerd
+
+For each container in the pod, a cgroup is created within the pod cgroup (ie, under `/sys/fs/cgroup/*/kubepod/pod.*/`
+for a guaranteed pod). This is not necessary; only a single cgroup which constrains the hypervisor
+appropriately is required.
+
+##### Where are all the vCPUS
+
+The vCPUs are placed under the pause container:
```

Comment:
Kata tries to setup the pause cgroup to represent the VM itself. It scales its resources to match the total resources assigned to the pod.